prow/repoowners/repoowners_test.go

```diff
@@ -889,9 +889,8 @@ func testLoadRepoOwners(clients localgit.Clients, t *testing.T) {
 		},
 	}
 
-	for _, test := range tests {
-		// copy to scoped variable, get rid of loopclosure linting error
-		test := test
```

Comment:
the existing version of this actually works fine

Reply:
though I wouldn't object to this change, but FYI this is a relatively common and correct solution as-is

Reply:
It does fail to `loopclosure` without the previous fix (variable copying), and it is cleaner to fix all of them the same one-line way and then there is no need for the comment either.

Reply:
> It does fail to loopclosure without the previous fix (variable copying)

I can't reproduce this, the linters are passing locally and in CI.

This approach is pretty common and is documented by the loopclosured tool https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/loopclosure

Reply:
This https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/29305/pull-test-infra-verify-lint/1646925296026259456 is where #29305 failed in the start. Hence fixed those in #29315. Rebased #29305, and then it fails like this: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/29305/pull-test-infra-verify-lint/1649299311588020224  hence were here where fixing that loopclosure issue leads to finding unit test that is really bad and broken.

Reply:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/29305/pull-test-infra-verify-lint/1646925296026259456

is wrong though:
```go
	for _, test := range tests {
		// copy to scoped variable, get rid of loopclosure linting error
		test := test
```

does not have a loop closure issue, this does ensure that test is not shared between t.Run() calls. as documented by https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/loopclosure which describes using `foo := foo` itself.

test becomes a new variable shadowing test instead of shared memory at the loop scope. I'd actually say it's the most common variant and we've used it many times to fix bugs through the kubernetes org.

so while this change is _fine_, it's not great that it's being prompted by seemingly broken linter output, and that's my concern. we don't want the linter falsely labeling this as broken and then churning the whole repo on this.

You can test that this works by https://go.dev/play/p/VGNQrfgXGJ1

Reply:
Funny enough, even if the linter was incorrect labeling these as loop closure problem, it did highlight them which led to #29315 which (as pointed by your playground example) fixed the tests, and to this one which fixes the 3rd one.

Now, it seems the best action would be to just remove the misleading comment from these 2 tests, and leave the variable copy in and not modify for loop. Third fix in trick-or-treat should be fine as it is in this PR. Correct?

Decision if trick-or-treat should be removed or this test removed is then left to be discussed among  maintainers.